### PR TITLE
Handle StorageLookupError exceptions

### PR DESCRIPTION
**Pull request by mbarnes**
This catches `StorageLookupError` exceptions where appropriate, with a couple other bug fixes and debugging enhancements thrown in.

Still investigating some failing E2E test cases, but with these commits I'm down to 4.

### CODE REVIEW

**Comment by mbarnes** [comment]
[(3) [multi-host-mgr-2] Object Not Found Exceptions](https://trello.com/c/SkTNgqMl/627-3-multi-host-mgr-2-object-not-found-exceptions)

**Comment by mbarnes** [comment]
[(3) [multi-host-mgr-2] Object Not Found Exceptions](https://trello.com/c/SkTNgqMl/627-3-multi-host-mgr-2-object-not-found-exceptions)

**Comment by mbarnes** [comment]
[(3) [multi-host-mgr-2] Object Not Found Exceptions](https://trello.com/c/SkTNgqMl/627-3-multi-host-mgr-2-object-not-found-exceptions)

**Comment by mbarnes** [comment]
[(3) [multi-host-mgr-2] Object Not Found Exceptions](https://trello.com/c/SkTNgqMl/627-3-multi-host-mgr-2-object-not-found-exceptions)

**Comment by mbarnes** [comment]
[(3) [multi-host-mgr-2] Object Not Found Exceptions](https://trello.com/c/SkTNgqMl/627-3-multi-host-mgr-2-object-not-found-exceptions)

**Comment by mbarnes** [comment]
[(3) [multi-host-mgr-2] Object Not Found Exceptions](https://trello.com/c/SkTNgqMl/627-3-multi-host-mgr-2-object-not-found-exceptions)

**Comment by mbarnes** [comment]
[(3) [multi-host-mgr-2] Object Not Found Exceptions](https://trello.com/c/SkTNgqMl/627-3-multi-host-mgr-2-object-not-found-exceptions)

**Comment by mbarnes** [comment]
[(3) [multi-host-mgr-2] Object Not Found Exceptions](https://trello.com/c/SkTNgqMl/627-3-multi-host-mgr-2-object-not-found-exceptions)

**Comment by mbarnes** [comment]
[(3) [multi-host-mgr-2] Object Not Found Exceptions](https://trello.com/c/SkTNgqMl/627-3-multi-host-mgr-2-object-not-found-exceptions)

**Comment by mbarnes** [comment]
[(3) [multi-host-mgr-2] Object Not Found Exceptions](https://trello.com/c/SkTNgqMl/627-3-multi-host-mgr-2-object-not-found-exceptions)

**Comment by mbarnes** [comment]
[(3) [multi-host-mgr-2] Object Not Found Exceptions](https://trello.com/c/SkTNgqMl/627-3-multi-host-mgr-2-object-not-found-exceptions)

**Comment by mbarnes** [comment]
[(3) [multi-host-mgr-2] Object Not Found Exceptions](https://trello.com/c/SkTNgqMl/627-3-multi-host-mgr-2-object-not-found-exceptions)

**Comment by mbarnes** [comment]
[(3) [multi-host-mgr-2] Object Not Found Exceptions](https://trello.com/c/SkTNgqMl/627-3-multi-host-mgr-2-object-not-found-exceptions)

**Comment by mbarnes** [comment]
[(3) [multi-host-mgr-2] Object Not Found Exceptions](https://trello.com/c/SkTNgqMl/627-3-multi-host-mgr-2-object-not-found-exceptions)

**Comment by mbarnes** [comment]
[(3) [multi-host-mgr-2] Object Not Found Exceptions](https://trello.com/c/SkTNgqMl/627-3-multi-host-mgr-2-object-not-found-exceptions)

**Comment by mbarnes** [comment]
[(3) [multi-host-mgr-2] Object Not Found Exceptions](https://trello.com/c/SkTNgqMl/627-3-multi-host-mgr-2-object-not-found-exceptions)

**Comment by mbarnes** [comment]
[(3) [multi-host-mgr-2] Object Not Found Exceptions](https://trello.com/c/SkTNgqMl/627-3-multi-host-mgr-2-object-not-found-exceptions)

File: src/commissaire_http/dispatcher/__init__.py [hunk 1]
```diff
@@ -154,11 +155,13 @@ def _get_params(self, environ, route, route_data):
 
         # If we are a PUT or POST look for params in wsgi.input
         if environ['REQUEST_METHOD'] in ('PUT', 'POST'):
-            if environ.get('CONTENT_LENGTH') and environ['CONTENT_LENGTH']:
+            content_length = int(environ.get('CONTENT_LENGTH', 0))
```
Comment (ashcrow) [inline src/commissaire_http/dispatcher/__init__.py]: :+1:

File: src/commissaire_http/dispatcher/__init__.py [hunk 2]
```diff
@@ -154,11 +155,13 @@ def _get_params(self, environ, route, route_data):
...
+            if content_length > 0:
                 try:
-                    params.update(json.loads(environ['wsgi.input'].read(
-                        int(environ['CONTENT_LENGTH'])).decode()))
-                except json.decoder.JSONDecodeError as error:
+                    wsgi_input = environ['wsgi.input'].read(content_length)
+                    more_params = eval(wsgi_input.decode())
+                    params.update(more_params)
+                except (ValueError, json.decoder.JSONDecodeError) as error:
```
Comment (ashcrow) [inline src/commissaire_http/dispatcher/__init__.py]: I like the change here but it just came to mind that we may want to throw an ISE here if data was given and we can not decode it. Thoughts?

Comment (mbarnes) [inline src/commissaire_http/dispatcher/__init__.py]: The key change here was adding the `eval()`, the rest of it was just breaking things up so I could debug it easier.

Comment (mbarnes) [inline src/commissaire_http/dispatcher/__init__.py]: Come to think of it, was that supposed to be JSON?  I just caught it trying to pass a dictionary-like string like `{'a': 123, 'b': 456}` to `params.update()`.

Comment (ashcrow) [inline src/commissaire_http/dispatcher/__init__.py]: It should be json. There isn't any middleware converting it.

Comment (ashcrow) [inline src/commissaire_http/dispatcher/__init__.py]: Looks like wsgiref does doe some work if the headers are provided. The results I see are like so:

## Without Headers

```python
b"{'a': 123, 'b': 456}"'
```

## With application/json

```python
b"{'a': 123, 'b': 456}"
```

**Pull request merged**